packages/react-client/src/screenShareTrackManager.ts

```diff
@@ -29,12 +29,37 @@ export const useScreenShare = <PeerMetadata, TrackMetadata>(
     setState({ stream, trackIds: { videoId, audioId } });
   };
 
+  const replaceTracks = async (newVideoTrack: MediaStreamTrack, newAudioTrack: MediaStreamTrack | null) => {
+    if (!state?.stream) return;
+
+    const addTrackPromises = [tsClient.replaceTrack(state.trackIds.videoId, newVideoTrack)];
+
+    if (newAudioTrack && state.trackIds.audioId) {
+      addTrackPromises.push(tsClient.replaceTrack(state.trackIds.audioId, newAudioTrack));
+    }
+
+    await Promise.all(addTrackPromises);
+  };
+
+  const setTracksMiddleware = async (middleware: TracksMiddleware | null): Promise<void> => {
+    if (!state?.stream) return;
+
+    const [videoTrack, audioTrack] = getTracks(state.stream);
+
+    if (!middleware) {
+      return await replaceTracks(videoTrack, audioTrack);
+    }
+
+    const [newVideoTrack, newAudioTrack] = middleware(videoTrack, audioTrack);
+    await replaceTracks(newVideoTrack, newAudioTrack);
```

Comment:
Just an idea:
```
    const [newVideoTrack, newAudioTrack] =  middleware ? middleware(videoTrack, audioTrack) : [videoTrack, audioTrack]
    await replaceTracks(newVideoTrack, newAudioTrack);
```

Reply:
i like that and i will take it even further